docs/src/transforms.md

```diff
@@ -0,0 +1,96 @@
+# [Transforms](@id about-transforms)
+
+A `Transform` defines a transformation of data, for example scaling, periodic functions, linear combination, one-hot encoding, etc.
+
+```@meta
+DocTestSetup = quote
+    using DataFrames
+    using Dates
+    using FeatureTransforms
+end
+```
+
+Usually, a `Transform` has one or more parameters. For example, we can define a squaring operation (i.e. raise to the power of 2):
+
+```jldoctest transforms
+julia> p = Power(2);
+```
+
+This transformation can be applied to data `x` as follows:
+
+```jldoctest transforms
+julia> x = [1.0, 2.0, 3.0];
+
+julia> FeatureTransforms.apply(x, p)
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+```
+
+## Applying a `Transform`
+
+There are three main ways to apply a `Transform` - suppose it is called `t`:
+
+* `Transforms.apply(data, t; kwargs...)` is non-mutating, returning the transformed data without modifying the original data.
+* `t(data; kwargs...)` is equivalent to `Transforms.apply(data, t; kwargs...)`
+* `Transforms.apply!(data, t; kwargs...)` is mutating, returning the modified `data` with the same type.
```

Comment:
Instead of laying this out in a way akin to a docstring, why not demonstrate it by expanding on the example above?